lib/oauth2/strategy/assertion.rb

```diff
@@ -4,19 +4,21 @@ module OAuth2
   module Strategy
     # The Client Assertion Strategy
     #
-    # @see http://tools.ietf.org/html/draft-ietf-oauth-v2-10#section-4.1.3
+    # @see https://tools.ietf.org/html/rfc7523
     #
     # Sample usage:
     #   client = OAuth2::Client.new(client_id, client_secret,
-    #                               :site => 'http://localhost:8080')
+    #                               :site => 'http://localhost:8080',
+    #                               :auth_scheme => :request_body)
     #
-    #   params = {:hmac_secret => "some secret",
-    #             # or :private_key => "private key string",
-    #             :iss => "http://localhost:3001",
-    #             :prn => "me@here.com",
-    #             :exp => Time.now.utc.to_i + 3600}
+    #   claimset = {
+    #     :iss => "http://localhost:3001",
+    #     :aud => "http://localhost:8080/oauth2/token"
+    #     :sub => "me@example.com",
+    #     :exp => Time.now.utc.to_i + 3600
+    #   }
     #
-    #   access = client.assertion.get_token(params)
+    #   access = client.assertion.get_token(claimset, :algorithm => 'HS256', :key => 'secret_key')
```

Comment:
I would prefer to have the second hash param be explicitly surrounded with curly braces, or setup as an instance variable like claimset.

This is because the presumed hash args without explicit curly braces have ways of badly breaking if the claimset hash gets inlined, or other parts become variables.

It isn't immediately clear from the documentation as is that this will not work:
```
algo = {:algorithm => 'HS256'}
key = {:key => 'secret_key'}
access = client.assertion.get_token(claimset, algo, key)
=> Boom!
```

Once we drop old Rubies we can avoid all this ambiguity and move to named parameters!

For now:
```
access = client.assertion.get_token(claimset, { :algorithm => 'HS256', :key => 'secret_key' })
```